tests/i18n/override.py

```diff
@@ -0,0 +1,168 @@
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with wildcard.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='*')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='gl')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) == 'gl'
+
+    # Test cookie with longer supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='en-US')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with longer underscore language code for a supported
+    # language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='gl_ES')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with longer hyphen language code for a supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='fr-FR')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) == 'fr'
+
+    # Test header with shorter language code for a supported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='en')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+    # Test header with unsupported language.
+    request = rf.get("", HTTP_ACCEPT_LANGUAGE='FAIL')
+    assert get_lang_from_http_header(request, SUPPORTED_LANGUAGES) is None
+
+
+@pytest.mark.django_db
+def test_get_lang_from_session(rf, client):
+    # Test no session.
+    request = rf.get("")
+    assert not hasattr(request, 'session')
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) is None
+
+    # Test session with no language.
+    response = client.get("")
+    request = response.wsgi_request
+    assert LANGUAGE_SESSION_KEY not in request.session
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) is None
+
+    # Test session with supported language.
+    response = client.get("")
+    request = response.wsgi_request
+    request.session[LANGUAGE_SESSION_KEY] = 'gl'
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) == 'gl'
+
+    # Test session with supported language.
+    response = client.get("")
+    request = response.wsgi_request
+    request.session[LANGUAGE_SESSION_KEY] = 'en-US'
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) == 'en-US'
+
+    # Test cookie with longer underscore language code for a supported
+    # language.
+    response = client.get("")
+    request = response.wsgi_request
+    request.session[LANGUAGE_SESSION_KEY] = 'gl_ES'
+
+    # Test cookie with longer hyphen language code for a supported language.
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) is None
+    response = client.get("")
+    request = response.wsgi_request
+    request.session[LANGUAGE_SESSION_KEY] = 'fr-FR'
+    assert get_lang_from_session(request, SUPPORTED_LANGUAGES) is None
```

Comment:
should this not fallback to `fr` ?

Reply:
Probably yes, but will need to change the function to support that. Will look into it.

Reply:
Updated PR to handle this.